cypress-tests/cypress/support/commands/projectsV2.ts

```diff
@@ -0,0 +1,77 @@
+import { TIMEOUTS } from "../../../config";
+
+export type ProjectIdentifier = {
+  slug: string;
+  namespace?: string;
+};
+
+interface NewProjectV2Props extends ProjectIdentifier {
+  visibility?: "public" | "private";
+  name: string;
+}
+
+function getUserNamespaceAPIV2(): Cypress.Chainable<string | null> {
+  return cy.request({ failOnStatusCode: false, method: "GET", url: `api/data/namespaces?minimum_role=editor` })
+    .then((response) => {
+      if (response.status === 200) {
+        const userNamespace = response.body?.filter((namespace) => namespace.namespace_kind === "user");
+        return userNamespace && userNamespace.length > 0 ? userNamespace[0].slug : null;
+      }
+      return null;
+    });
+}
+function createProjectIfMissingAPIV2(newProjectProps: NewProjectV2Props) {
+  cy.request({failOnStatusCode: false, method: "GET", url: `api/data/projects/${newProjectProps.namespace}/${newProjectProps.slug}`})
+    .then((response) => {
+      if (response.status != 200) {
+        cy.request({
+          method: "POST",
+          url: "api/data/projects",
+          body: newProjectProps,
+          headers: {
+            'Content-Type': 'application/json'
+          }
+        }).then((postResponse) => {
+          cy.visit(`v2/projects/${newProjectProps.namespace}/${newProjectProps.slug}`);
+          cy.get(`[data-cy="project-name"]`, { timeout: TIMEOUTS.vlong }).should(
+            "be.visible"
+          );
+        });
+    } else {
+      cy.visit(`v2/projects/${newProjectProps.namespace}/${newProjectProps.slug}`);
+      cy.get(`[data-cy="project-name"]`, { timeout: TIMEOUTS.vlong }).should(
+        "be.visible"
+      );
+    }
+  });
+}
+function deleteProjectFromAPIV2(id: ProjectIdentifier) {
+  cy.request({failOnStatusCode: false, method: "GET", url: `api/data/projects/${id.namespace}/${id.slug}`})
+    .then((response) => {
+      const projectId = response.body["id"];
+      if (response.status == 200) {
```

Comment:
suggestion: You can eliminate this if statement and the call to the GET method if you want because deleting a non-existing project should return a 204 as well. But you can check for a 204 or 404 in the call to the DELETE endpoint. And both of those will be acceptable status codes.

Alternatively you can also flip the order and remove the if statement. I.e. first call DELETE and make sure it is 204 or 404 and then call GET on the project id and make sure it is 404.